routing/bandwidth.go

```diff
@@ -89,30 +130,129 @@ func (b *bandwidthManager) getBandwidth(cid lnwire.ShortChannelID,
 		return 0
 	}
 
-	// If our link isn't currently in a state where it can  add another
+	// bandwidthResult is an inline type that we'll use to pass the
+	// bandwidth result from the external traffic shaper to the main logic
+	// below.
+	type bandwidthResult struct {
+		// bandwidth is the available bandwidth for the channel as
+		// reported by the external traffic shaper. If the external
+		// traffic shaper is not handling the channel, this value will
+		// be fn.None
+		bandwidth fn.Option[lnwire.MilliSatoshi]
+
+		// htlcAmount is the amount we're going to use to check if we
+		// can add another HTLC to the channel. If the external traffic
+		// shaper is handling the channel, we'll use 0 to just sanity
+		// check the number of HTLCs on the channel, since we don't know
+		// the actual HTLC amount that will be sent.
```

Comment:
We don't know the HTLC amount yet because the external traffic shaper might want to send some of the value of this HTLC in a different form (e.g. assets). So the actual BTC amount that goes on-chain might be just above dust, but the actually transported "worth" might be millions of satoshis.